### ISSUE TITLE
multi_scan expects `xcresult` in output_types, but scan doesn't

### ISSUE THREAD

**Issue by lilyball** [lyndsey-ferguson/fastlane-plugin-test_center#253]
### New Issue Checklist

- [X] Updated `fastlane-plugin-test_center` to the latest version
- [X] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [X] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [X] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

> If you love this fastlane plugin, consider sponsoring it or asking your company to sponsor it. I would really appreciate any
> gesture: https://github.com/sponsors/lyndsey-ferguson. 😍

### Issue Description
When I pass `output_types: "html,junit"` I don't get an xcresult bundle. When I pass `output_types: "html,junit,xcresult"` I do, but `scan` emits a warning about how `xcresult` isn't a known output type. It seems that `scan` wants me to just pass `result_bundle: true` to get the xcresult, but when I give that to `multi_scan` it prints a warning about how the "test_result" bundle will be a symlink to the xcresult, and assuming collation actually happens (e.g. assuming I actually have more than one batch or simulator) then I do in fact get a symlink.

This is on Xcode 11 of course.

Here's what I think test_center should do:

* Don't add the symlink when I provide `result_bundle: true`. `scan` doesn't do this, why should `multi_scan`?
* Given the fact that existing callers are passing `xcresult` as an output type and living with the warning (we're certainly doing that so I assume others are too), test_center should strip that from the `output_types` it passes to `scan`.

It's also entirely possible I'm misunderstanding something here, but in my experiments with `scan`, if I pass `output_types: "html,junit"` I don't get an xcresult bundle without `result_bundle: true`. Digging through the test_center code it looks like test_center is actually trying to strip `result_bundle` from the parameters, and if true then I don't know how it's getting an xcresult from scan, unless it's digging through the Derived Data since there seems to be an .xcresult bundle in there.

Ultimately, even after doing all this testing and looking through code, I'm still confused as to what test_center wants me to do.

**Comment by lyndsey-ferguson** [comment]
So, the option to pass `xcresult` to `:output_types` was something that I added while `scan` was still writing out `result_bundles`. I didn't like that there were two different ways to get these types of output types, so I added `xcresult`.

`scan` then changed the file name of the result bundle to use the `xcresult` file extension. That broke what people were expecting from `multi_scan` so I tweaked the code to create a symbolic link, "result_bundle" to the actual `xcresult` file.

In order to control what the xcresult bundle name is called, I update the `xcargs` myself to provide the desired `xcresult` bundle name rather than relying on what `scan` provides. This mean that I remove the `:result_bundle: true` option when calling `scan` directly.

I can stop passing `xcresult` as an option for `:output_types` to `scan` and remove that warning. I can also improve the documentation around this so that it is clearer what is expected and what is happening.

**Comment by lyndsey-ferguson** [comment]
@lilyball, I have fixed the warning from `scan` regarding the `xcresult` output type. Can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-253-xcresult-confusion"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Additionally**: I have added some [documentation](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/pull/259/files?short_path=c9e7035#diff-c9e70358d3ab6fd9144199178d594d01) around the output type, can you take a look and let me know if it helps?

**Comment by lyndsey-ferguson** [comment]
@lilyball I would appreciate it if you could test this: I'd like to confirm that this addresses your concerns. If you're too busy right now, I can merge this on Friday and we can open up another issue if there are remaining problems.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.11.6](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.11.6)
